solidity/contracts/cross-chain/wormhole/L1BTCDepositorNttWithExecutor.sol

```diff
@@ -0,0 +1,648 @@
+        external
+        onlyOwner
+    {
+        require(
+            _newNttManagerWithExecutor != address(0),
+            "Address cannot be zero"
+        );
+        address oldManager = address(nttManagerWithExecutor);
+        nttManagerWithExecutor = INttManagerWithExecutor(
+            _newNttManagerWithExecutor
+        );
+        emit NttManagerWithExecutorUpdated(
+            oldManager,
+            _newNttManagerWithExecutor
+        );
+    }
+
+    /// @notice Allows the owner to retrieve tokens from the contract
+    /// @param _token The address of the token to retrieve (address(0) for native token)
+    /// @param _to The address to send the tokens to
+    /// @param _amount The amount of tokens to retrieve
+    function retrieveTokens(
+        address _token,
+        address _to,
+        uint256 _amount
+    ) external onlyOwner {
+        require(
+            _to != address(0),
+            "Cannot retrieve tokens to the zero address"
+        );
+
+        if (_token == address(0)) {
+            payable(_to).transfer(_amount);
+        } else {
+            IERC20Upgradeable(_token).safeTransfer(_to, _amount);
+        }
+    }
+
+    /// @notice Quotes the cost using stored executor parameters
+    /// @dev Must call setExecutorParameters() first with real signed quote
+    /// @return cost Total cost for the transfer using stored parameters
+    function quoteFinalizeDeposit() external view returns (uint256 cost) {
+        require(
+            executorParametersSet,
+            "Must call setExecutorParameters() first with real signed quote"
+        );
+
+        // Extract destination chain from stored executor parameters
+        // We'll use the default chain since we don't have the receiver encoded in stored params
+        uint16 defaultChain = _getDefaultSupportedChain();
+        require(defaultChain != 0, "No supported chains configured");
+
+        return
+            nttManagerWithExecutor.quoteDeliveryPrice(
+                underlyingNttManager,
+                defaultChain,
+                "", // Empty transceiver instructions for basic transfer
+                storedExecutorArgs,
+                storedFeeArgs
+            );
+    }
+
+    /// @notice Sets executor parameters for the next finalizeDeposit call
+    /// @param executorArgs Real executor arguments with valid signed quote from Wormhole Executor API
+    /// @param feeArgs Fee arguments for the executor service
+    /// @dev Must be called before finalizeDeposit() to provide real signed quote
+    function setExecutorParameters(
+        ExecutorArgs memory executorArgs,
+        FeeArgs memory feeArgs
+    ) external {
+        // CRITICAL: Validate that we have a real signed quote
+        require(
+            executorArgs.signedQuote.length > 0,
+            "Real signed quote from Wormhole Executor API is required"
+        );
+        _validateSignedQuote(executorArgs.signedQuote);
```

Comment:
Looks like the quote is validated, is this TODO outdated? Or does it refer to something else?

Reply:
this only checks the format of the quote, will change the name to be more clear, is not a price validation just the data format.

Reply:
`_validateSignedQuote` → `_validateSignedQuoteFormat`